solr/core/src/test/org/apache/solr/security/JWTAuthPluginIntegrationTest.java

```diff
@@ -70,54 +89,43 @@
-        .withDefaultClusterProperty("useLegacyReplicaAssignment", "false")
-        .configure();
-    baseUrl = cluster.getRandomJetty(random()).getBaseUrl().toString();
+  private static MockOAuth2Server mockOAuth2Server;
 
-    String jwkJSON = "{\n" +
-        "  \"kty\": \"RSA\",\n" +
-        "  \"d\": \"i6pyv2z3o-MlYytWsOr3IE1olu2RXZBzjPRBNgWAP1TlLNaphHEvH5aHhe_CtBAastgFFMuP29CFhaL3_tGczkvWJkSveZQN2AHWHgRShKgoSVMspkhOt3Ghha4CvpnZ9BnQzVHnaBnHDTTTfVgXz7P1ZNBhQY4URG61DKIF-JSSClyh1xKuMoJX0lILXDYGGcjVTZL_hci4IXPPTpOJHV51-pxuO7WU5M9252UYoiYyCJ56ai8N49aKIMsqhdGuO4aWUwsGIW4oQpjtce5eEojCprYl-9rDhTwLAFoBtjy6LvkqlR2Ae5dKZYpStljBjK8PJrBvWZjXAEMDdQ8PuQ\",\n" +
-        "  \"e\": \"AQAB\",\n" +
-        "  \"use\": \"sig\",\n" +
-        "  \"kid\": \"test\",\n" +
-        "  \"alg\": \"RS256\",\n" +
-        "  \"n\": \"jeyrvOaZrmKWjyNXt0myAc_pJ1hNt3aRupExJEx1ewPaL9J9HFgSCjMrYxCB1ETO1NDyZ3nSgjZis-jHHDqBxBjRdq_t1E2rkGFaYbxAyKt220Pwgme_SFTB9MXVrFQGkKyjmQeVmOmV6zM3KK8uMdKQJ4aoKmwBcF5Zg7EZdDcKOFgpgva1Jq-FlEsaJ2xrYDYo3KnGcOHIt9_0NQeLsqZbeWYLxYni7uROFncXYV5FhSJCeR4A_rrbwlaCydGxE0ToC_9HNYibUHlkJjqyUhAgORCbNS8JLCJH8NUi5sDdIawK9GTSyvsJXZ-QHqo4cMUuxWV5AJtaRGghuMUfqQ\"\n" +
-        "}";
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    // Setup an OAuth2 mock server with SSL
+    Path p12Cert = SolrTestCaseJ4.TEST_PATH().resolve("security").resolve("jwt_plugin_idp_certs.p12");
+    pemFilePath = SolrTestCaseJ4.TEST_PATH().resolve("security").resolve("jwt_plugin_idp_cert.pem");
+    wrongPemFilePath = SolrTestCaseJ4.TEST_PATH().resolve("security").resolve("jwt_plugin_idp_wrongcert.pem");
 
-    PublicJsonWebKey jwk = RsaJsonWebKey.Factory.newPublicJwk(jwkJSON);
-    JwtClaims claims = JWTAuthPluginTest.generateClaims();
-    jws = new JsonWebSignature();
-    jws.setPayload(claims.toJson());
-    jws.setKey(jwk.getPrivateKey());
-    jws.setKeyIdHeaderValue(jwk.getKeyId());
-    jws.setAlgorithmHeaderValue(AlgorithmIdentifiers.RSA_USING_SHA256);
+    mockOAuth2Server = createMockOAuthServer(p12Cert,"secret");
+    mockOAuth2Server.start();
+    mockOAuthToken = mockOAuth2Server.issueToken("default",
+        "myClientId",
+        new DefaultOAuth2TokenCallback()).serialize();
+    initStaticJwt();
+  }
 
-    jwtTestToken = jws.getCompactSerialization();
-    
-    PublicJsonWebKey jwk2 = RsaJwkGenerator.generateJwk(2048);
-    jwk2.setKeyId("k2");
-    JsonWebSignature jws2 = new JsonWebSignature();
-    jws2.setPayload(claims.toJson());
-    jws2.setKey(jwk2.getPrivateKey());
-    jws2.setKeyIdHeaderValue(jwk2.getKeyId());
-    jws2.setAlgorithmHeaderValue(AlgorithmIdentifiers.RSA_USING_SHA256);
-    jwtTokenWrongSignature = jws2.getCompactSerialization();
+  @AfterClass
+  public static void afterClass() throws Exception {
+    if (mockOAuth2Server != null) {
+      mockOAuth2Server.shutdown();
+    }
+  }
 
-    cluster.waitForAllNodes(10);
+  @Override
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
```

Comment:
if all it does it call super, then we don't need this method.

Reply:
Agree, removed it.